### DISCUSSION TITLE
FirefoxでGIFアニメが動作していない

### DISCUSSION THREAD

**Discussion by ryo0301** [increments/qiita-discussions#422] [要望・アイデア・不具合]
FirefoxでQiitaの記事を参照した時に、GIFアニメが止まったままでアニメーションしません。
記事投稿時はアニメーションしていたことは確認しています。
画像をクリックして画像のみを表示するとアニメーションします。

記事内の画像形式はAVIF、クリック後の画像形式はGIFでした。
Chromeでは記事内の画像もアニメーションしました。

参照した記事: https://qiita.com/ryo0301/items/506c4ed042d74b8f8987
OS: macOS 12.6.2
Firefox: 109.0

**Comment by kyntk** [comment]
@ryo0301 ご報告ありがとうございます！

AVIFは、Chromeではanimationが動作する一方、Firefoxではstaticはサポートしているもののanimationがサポートされていません。
https://caniuse.com/avif

現在、CDNにおいてブラウザが対応している形式に合わせて適切なファイルを返却しているのですが、Firefoxのリクエストヘッダーを確認すると、AVIFをacceptするという設定になっています。
この時、CDNの仕様でstatic AVIFが返却されるようになっています。

そのため、https://github.com/increments/qiita-discussions/discussions/196 でも同様にご意見をいただいているのですが、すぐには解消が難しそうです。
ご不便をおかけして申し訳ないですが、ご了承いただけますと幸いです🙇

**Reply by Phroneris** [reply]
Windows 10のFirefox 109.0.1でも、確かにこちらの「参照した記事」のGIFアニメは全く動きませんね。
一方で #196 の記事は、そこの追記にもあるように同じ環境で全てのGIFアニメが動いて見られました。
動く場合と動かない場合があるのでしょうか？ それはそれで厄介ですね。

記事なりコメントなりの投稿者は基本的に、動くつもりでGIFアニメを掲載しているはずです。
それが一部のメジャーな環境では動く見込みが長期的に無く、投稿者に対する「一部のメジャーな環境で動かないことがあります」のような警告も特に無く、読者に対する「これはもし動いてなくても動くはずの画像なんです」のような示唆も別に無いというのは、ちょっと…かなり不親切に感じます。

投稿者側にできることとして、「動かない環境も想定して1フレーム目を静止画として適切なものにする」のような防衛策[^1]は考えられますが…。
APNGのような比較的新しいアニメーション形式なら色々な人への配慮としてそういう技法がある程度共有されているものの、『ポケモンショック』より古い時代から動くものとして普及しているGIFアニメ形式では逆に盲点だったりするんじゃないでしょうか。
そもそも投稿者は何らかの技術者ではあってもグラフィック方面の有識者とは限りませんから（私も違いますが）、このような方策が一般的に期待されるべき常識レベルのものだとはどうも思えません。

読者側なら尚更、知ったこっちゃ無い話です。
「なんかこの記事謎の画像置いてて意味わからんな」と読者が投稿者への心証を下げるような事案が、実はサービスの技術的不具合によって起きている、というストーリーが十分想定できてしまうと思うのですが、どうでしょうか。

ユーザーとして、この現状は了承しづらいです。
動かないものは動かないのでそれは仕方無いのですが、それなら投稿者と読者の双方にそれを正しく警戒させるべきではないかと思います。
もちろん動けば最高です。

[^1]: 検索とかからここに辿り着いてこれを初めて知ったそこのお前！ こうやってアニメーション画像を防衛していこうな！

**Reply by tanaka-asami415** [reply]
@Phroneris 
ご意見ありがとうございます🙏
Firefox側の仕様が原因ということもあり、すぐには解消が難しいため
一旦、ヘルプやMarkdown記法 チートシートに記載するようにいたします🙇

**Reply by tanaka-asami415** [reply]
@tsubolabo 
ありがとうございます！参考にさせていただきます🙏

**Reply by esperecyan** [reply]
[Firefox 113.0](https://www.mozilla.org/firefox/113.0/releasenotes/ "Firefox 113.0, See All New Features, Updates and Fixes")で対応したようです。
2023-09-26のFirefox 102 ESRのサポート終了をもって、Firefox ESRでも問題が解消する見込みです。
https://wiki.mozilla.org/Release_Management/Calendar

**Reply by Phroneris** [reply]
@esperecyan
ｳｵｰｰｯほんとだ、最新版に更新したら動くようになってました。
お知らせくださりありがとうございます！

@tanaka-asami415
#196 にある記事のように旧版Firefoxでも動く場合があったというのは、どういう原理なのでしょうか？

**Reply by kyntk** [reply]
@esperecyan
本当ですね、ご報告ありがとうございます！

@Phroneris
解消されてよかったですね！ ✨ 

GIFファイルは、記事の投稿後には最適なフォーマットが選択されて配信していたため、AVIFが返却されていますが、
編集ページのプレビューではGIFのままプレビューが表示されているため動いていました。
また、記事投稿後であっても、最適化のための設定が行われていないことがあり、GIFのまま配信されていたものもあったようです :face_with_spiral_eyes:
